server/publications/collections/products.js

```diff
@@ -347,7 +356,7 @@ Meteor.publish("Products", function (productScrollLimit = 24, productFilters, so
       $in: [true, false, null, undefined]
     };
     selector.shopId = {
-      $in: userAdminShopIds
+      $in: _.intersection(userAdminShopIds, activeShopsIds)
```

Comment:
for the Primary Shop, this will have little impact, however, for Merchant Shops, owners will only see their products.
Again, probably not a huge impact because Merchant Shop Owners are unlikely to have createProduct roles in other shops.